Cubical/Categories/Instances/PosetCat/Monotone.agda

```diff
@@ -0,0 +1,207 @@
+
+  Sigma≡ : {X Y : Poset ℓ ℓ'} -> {s1 s2 : Sigma X Y} ->
+    s1 .fst ≡ s2 .fst -> s1 ≡ s2
+  Sigma≡ {X} {Y} = Σ≡Prop (λ f → isPropΠ3
+    (λ x y x≤y -> IsPoset.is-prop-valued (PosetStr.isPoset (str Y)) (f x) (f y)))
+
+  -- Equality of monotone functions is equivalent to equality of the
+  -- underlying functions.
+  eqMon' : {X Y : Poset ℓ ℓ'} -> (f g : MonFun' X Y) ->
+    MonFun'.f f ≡ MonFun'.f g -> f ≡ g
+  eqMon' {X} {Y} f g p = isoFunInjective MonFun'IsoΣ f g
+    (Σ≡Prop (λ h → isPropΠ3 (λ y z q → IsPoset.is-prop-valued (PosetStr.isPoset (str Y)) (h y) (h z))) p)
+
+  eqMon : {X Y : Poset ℓ ℓ'} -> (f g : MonFun X Y) ->
+    MonFun.f f ≡ MonFun.f g -> f ≡ g
+  eqMon {X} {Y} f g p = isoFunInjective isoMonFunMonFun' f g (eqMon' _ _ p)
+
+  -- isSet for Sigma
+  isSetSigma : {X Y : Poset ℓ ℓ'} -> isSet (Sigma X Y)
+  isSetSigma {X} {Y} = isSetΣSndProp
+    (isSet→ (IsPoset.is-set (PosetStr.isPoset (str Y))))
+    λ f -> isPropIsMon' {X} {Y} f
+
+  -- isSet for monotone functions
+  MonFunIsSet : {X Y : Poset ℓ ℓ'} -> isSet (MonFun X Y)
+  MonFunIsSet {X} {Y} = let composedIso = (compIso isoMonFunMonFun' MonFun'IsoΣ) in
+    isSetRetract
+      (Iso.fun composedIso) (Iso.inv composedIso) (Iso.leftInv composedIso)
+      (isSetΣSndProp
+        (isSet→ (IsPoset.is-set (PosetStr.isPoset (str Y))))
+        (isPropIsMon' {X} {Y}))
+
+
+
+ -- Ordering on monotone functions
+  module _ {X Y : Poset ℓ ℓ'} where
+
+    _≤mon_ :
+      MonFun X Y → MonFun X Y → Type (ℓ-max ℓ ℓ')
+    _≤mon_ f g = ∀ x -> (PosetStr._≤_ (Y .snd)) (MonFun.f f x) (MonFun.f g x)
+
+    ≤mon-prop : isPropValued _≤mon_
+    ≤mon-prop f g =
+      isPropΠ (λ x -> (IsPoset.is-prop-valued (PosetStr.isPoset (str Y))) (MonFun.f f x) (MonFun.f g x))
+
+    ≤mon-refl : isRefl _≤mon_
+    ≤mon-refl = λ f x → (IsPoset.is-refl (PosetStr.isPoset (str Y))) (MonFun.f f x)
+
+    ≤mon-trans : isTrans _≤mon_
+    ≤mon-trans = λ f g h f≤g g≤h x →
+      (IsPoset.is-trans (PosetStr.isPoset (str Y))) (MonFun.f f x) (MonFun.f g x) (MonFun.f h x)
+        (f≤g x) (g≤h x)
+
+    ≤mon-antisym : isAntisym _≤mon_
+    ≤mon-antisym = λ f g f≤g g≤f → eqMon f g
+      (funExt λ x →
+        (IsPoset.is-antisym (PosetStr.isPoset (str Y))) (MonFun.f f x) (MonFun.f g x)
+          (f≤g x) (g≤f x)
+      )
+
+
+    -- Alternate definition of ordering on monotone functions, where we allow for the
+    -- arguments to be distinct
+    _≤mon-het_ : MonFun X Y -> MonFun X Y -> Type (ℓ-max ℓ ℓ')
+    _≤mon-het_ f f' = ∀ x x' -> (PosetStr._≤_ (X .snd)) x x' -> (PosetStr._≤_ (Y .snd)) (MonFun.f f x) (MonFun.f f' x')
+
+    ≤mon→≤mon-het : (f f' : MonFun X Y) -> f ≤mon f' -> f ≤mon-het f'
+    ≤mon→≤mon-het f f' f≤f' = λ x x' x≤x' →
+      MonFun.f f x    ≤⟨ MonFun.isMon f x≤x' ⟩
+      MonFun.f f x'   ≤⟨ f≤f' x' ⟩
+      MonFun.f f' x'  ◾
+      where
+        open PosetReasoning Y
+
+
+ -- Poset of monotone functions between two posets
+  IntHom : Poset ℓ ℓ' -> Poset ℓ ℓ' ->
```

Comment:
Should refactor this in terms of `Exponential`

Reply:
Held off on this as it looks like we need to define the category of posets first before defining the exponential for it. Didn't want to create a circular dependency with the definition. We should probably move this to a different file to refactor